recipes/nspr/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        if self.settings.os == "Windows":
+            if self.options.shared:
+                os.mkdir(os.path.join(self.package_folder, "bin"))
+            for lib in self._library_names:
+                libsuffix = "lib" if self.settings.compiler == "Visual Studio" else "a"
+                libprefix = "" if self.settings.compiler == "Visual Studio" else "lib"
+                if self.options.shared:
+                    os.unlink(os.path.join(self.package_folder, "lib", "{}{}_s.{}".format(libprefix, lib, libsuffix)))
+                    os.rename(os.path.join(self.package_folder, "lib", "{}.dll".format(lib)),
+                              os.path.join(self.package_folder, "bin", "{}.dll".format(lib)))
+                else:
+                    os.unlink(os.path.join(self.package_folder, "lib", "{}{}.{}".format(libprefix, lib, libsuffix)))
+                    os.unlink(os.path.join(self.package_folder, "lib", "{}.dll".format(lib)))
+            if not self.options.shared:
+                tools.replace_in_file(os.path.join(self.package_folder, "include", "nspr", "prtypes.h"),
+                                      "#define NSPR_API(__type) PR_IMPORT(__type)",
+                                      "#define NSPR_API(__type) extern __type")
+                tools.replace_in_file(os.path.join(self.package_folder, "include", "nspr", "prtypes.h"),
+                                      "#define NSPR_DATA_API(__type) PR_IMPORT_DATA(__type)",
+                                      "#define NSPR_DATA_API(__type) extern __type")
+        elif self.settings.os == "Macos":
+            for lib in self._library_names:
+                if self.options.shared:
+                    os.unlink(os.path.join(self.package_folder, "lib", "lib{}.a".format(lib)))
+                else:
+                    os.unlink(os.path.join(self.package_folder, "lib", "lib{}.dylib".format(lib)))
+
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.build_type == "Debug":
+                for lib in self._library_names:
+                    os.remove(os.path.join(self.package_folder, "lib", "{}.pdb".format(lib)))
+
+        if not self.options.shared or self.settings.os == "Windows":
+            for f in os.listdir(os.path.join(self.package_folder, "lib")):
+                os.chmod(os.path.join(self.package_folder, "lib", f), 0o644)
+
+    @property
+    def _library_names(self):
+        return ["plds4", "plc4", "nspr4"]
+
+    def package_info(self):
+        libs = self._library_names
+        if self.settings.os == "Windows" and not self.options.shared:
+            libs = list("{}_s".format(l) for l in libs)
+        self.cpp_info.libs = libs
+        if self.settings.arch == "x86":
+            self.cpp_info.defines.append("_M_IX86")
```

Comment:
is this really needed? seems to be automatically defined by MSVC compiler:
https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019

Reply:
These definitions are missing on mingw.
I've changed the recipe to conditionally add them.